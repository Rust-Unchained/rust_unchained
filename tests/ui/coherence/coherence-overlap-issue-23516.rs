

pub trait Sugar { fn dummy(&self) { } }
pub trait Sweet { fn dummy(&self) { } }
impl<T:Sugar> Sweet for T { }
impl<U:Sugar> Sweet for Box<U> { }

// Additional requirements for conflicts to happen in Unchained Rust
impl Sugar for i32 {}
impl Sweet for Box<i32> {}
//~^ ERROR E0119

fn main() { }
