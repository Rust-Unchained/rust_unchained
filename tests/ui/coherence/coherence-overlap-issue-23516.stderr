error[E0119]: conflicting implementations of trait `Sweet` for type `Box<i32>`
  --> $DIR/coherence-overlap-issue-23516.rs:10:1
   |
LL | impl<U:Sugar> Sweet for Box<U> { }
   | ------------------------------ first implementation here
...
LL | impl Sweet for Box<i32> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Box<i32>`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
